### DISCUSSION TITLE
Three Useful Functions in Base R: `regexec()`, `strrep()`, and `append()` | /en/2023/10/three-functions/

### DISCUSSION THREAD

**Discussion by giscus** [rbind/yihui#1528] [Comments]
I just finished reviewing a pull request in the knitr repo that tries to improve the error message when it fails to parse YAML, and I feel three base R functions are worth mentioning to more R users. &hellip;

https://yihui.org/en/2023/10/three-functions/

**Comment by maelle** [comment]
So cool! :smile_cat: I learnt about `strrep()` recently (it was on my sticky note, you beat me to it :grin:) by reading https://github.com/r-lib/lintr/issues/884 (it's in lintr now https://github.com/r-lib/lintr/pull/1652)

**Comment by YaoxiangLi** [comment]
Cool! I like this one: `x = c("Failed to parse YAML: ", x, "\n", append(meta, cursor, row))`

BTW, I can help on the stuffs if still needed
https://yihui.org/todo/#help-needed

**Reply by yihui** [reply]
Thank you! Please feel free to work on anything in the list, although I haven't updated this TODO list for a long time, and there are some other things that I'm interested in getting done.

**Reply by YaoxiangLi** [reply]
Understood. Please feel free to send me a more recent version or other tasks that you're interested in getting done.
